### DISCUSSION TITLE
Source code interpreter

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#129] [Q&A] [answered]
Objeck looks very like Ruby to me. Currently it follows the Java style, compiling to bytecode first then the interpreter will interpret the compiled bytecode but not the source. What about adding another interpreter (objeck.exe) able to interpret the source directly?

**Comment by ghost** [comment]
I think Objeck is in the same category as Java and is not a scripting language. If Objeck is like Java then the bytecode compiled by it is platform independent. Languages like Ruby and Python also compile to bytecode behind the scene but the bytecode is platform dependent.

**Comment by objeck** [comment, marked as answer]
Yes, it is possible now that the compiler output and VM input are read and written from tandem buffers. 

The compiler writes bytes into a zLib compressed buffer written to disk. The VM reads and decompresses the bytes and loads the program. Not too hard to exchange uncompressed buffers between the compiler and VM. 

That said, not something I have time to work on now. 

Something the community could implement?
